aloha-core/src/main/scala/com/eharmony/aloha/audit/impl/scoreproto/ScoreAuditor.scala

```
@@ -0,0 +1,254 @@
+        if (e.nonEmpty)
+          eb.addAllMessages(e)
+      }
+
+      val baseScore = boxScore(valueToAudit).setModel(m)
+      b.setScore(prob.fold(baseScore)(baseScore.setProbability))
+      b.build
+    }
+
+    private[this] def mId(modelId: ModelIdentity) =
+      ModelId.newBuilder.setId(modelId.getId()).setName(modelId.getName()).build
+
+    private[this] def accumulateMissing(missing: Iterable[String], subScores: Iterable[Score]) = {
+      val subMissing = subScores.flatMap(s => iterableAsScalaIterable(s.getError.getMissingFeatures.getNamesList))
+      val mm = missing ++ subMissing
+      if (mm.nonEmpty) mm.toSeq.distinct else Nil
+    }
+  }
+
+  //  private[this] object NothingAuditor extends ScoreAuditorImpl[Nothing] {
+  //    override def boxScore(valueToAudit: Nothing): BaseScore.Builder =
+  //      BaseScore.newBuilder.setType(ScoreType.NONE)
+  //  }
+
+  private[this] object BooleanAuditor extends ScoreAuditorImpl[Boolean] {
+    override def boxScore(valueToAudit: Boolean): BaseScore.Builder =
+      BaseScore.newBuilder.setType(ScoreType.BOOLEAN).setExtension(
+        BooleanScore.impl,
+        BooleanScore.newBuilder.setScore(valueToAudit).build
+      )
+
+    private[aloha] def unapply(value: Score): Option[Boolean] = {
+      val v = value.getScore.getExtension(BooleanScore.impl)
+      if (v.hasScore)
+        Some(v.getScore)
+      else None
+    }
+  }
+
+  private[this] object ByteAuditor extends ScoreAuditorImpl[Byte] {
+    override def boxScore(valueToAudit: Byte): BaseScore.Builder =
+      BaseScore.newBuilder.setType(ScoreType.INT).setExtension(
+        IntScore.impl,
+        IntScore.newBuilder.setScore(valueToAudit).build
+      )
+
+    private[aloha] def unapply(value: Score): Option[Byte] = {
+      val v = value.getScore.getExtension(IntScore.impl)
+      if (v.hasScore) {
+        val sc = v.getScore
+        if (Byte.MinValue <= sc && sc <= Byte.MaxValue)
+          Some(sc.toByte)
+        else None
```

Do we want to log this or propogate this somehow?  It seems like a silent error.